nucypher/blockchain/eth/agents.py

```diff
@@ -1153,159 +1162,160 @@ def _call_function_by_name(name: str):
+        return Wei(current_bid)
 
-    @property
+    @property  # type: ignore
     @contract_api(CONTRACT_ATTRIBUTE)
-    def lot_value(self) -> int:
+    def lot_value(self) -> NuNits:
         """
         Total number of tokens than can be bid for and awarded in or the number of NU
         tokens deposited before the bidding windows begins via tokenDeposit().
         """
-        supply = self.contract.functions.tokenSupply().call()
-        return supply
+        supply: int = self.contract.functions.tokenSupply().call()
+        return NuNits(supply)
 
     @contract_api(CONTRACT_CALL)
-    def get_bonus_lot_value(self) -> int:
+    def get_bonus_lot_value(self) -> NuNits:
         """
         Total number of tokens than can be  awarded for bonus part of bid.
         """
-        num_bidders = self.get_bidders_population()
-        supply = self.lot_value - num_bidders * self.contract.functions.minAllowableLockedTokens().call()
-        return supply
+        num_bidders: int = self.get_bidders_population()
+        supply: int = self.lot_value - num_bidders * self.contract.functions.minAllowableLockedTokens().call()
+        return NuNits(supply)
 
     @contract_api(CONTRACT_CALL)
-    def get_remaining_work(self, checksum_address: str) -> int:
+    def get_remaining_work(self, checksum_address: str) -> PeriodDelta:
         """Get remaining work periods until full refund for the target address."""
         result = self.contract.functions.getRemainingWork(checksum_address).call()
-        return result
+        return PeriodDelta(result)  # TODO: Double check this type
 
     @contract_api(CONTRACT_CALL)
-    def get_bonus_eth_supply(self) -> int:
+    def get_bonus_eth_supply(self) -> Wei:
         supply = self.contract.functions.bonusETHSupply().call()
-        return supply
+        return Wei(supply)
 
     @contract_api(CONTRACT_CALL)
-    def get_eth_supply(self) -> int:
-        num_bidders = self.get_bidders_population()
-        min_bid = self.minimum_allowed_bid
-        supply = num_bidders * min_bid + self.get_bonus_eth_supply()
-        return supply
+    def get_eth_supply(self) -> Wei:
+        num_bidders: int = self.get_bidders_population()
+        min_bid: int = self.minimum_allowed_bid
+        supply: int = num_bidders * min_bid + self.get_bonus_eth_supply()
+        return Wei(supply)
 
-    @property
+    @property  # type: ignore
     @contract_api(CONTRACT_ATTRIBUTE)
     def boosting_refund(self) -> int:
         refund = self.contract.functions.boostingRefund().call()
         return refund
 
-    @property
+    @property  # type: ignore
     @contract_api(CONTRACT_ATTRIBUTE)
     def slowing_refund(self) -> int:
-        refund = self.contract.functions.SLOWING_REFUND().call()
+        refund: int = self.contract.functions.SLOWING_REFUND().call()
         return refund
 
     @contract_api(CONTRACT_CALL)
-    def get_bonus_refund_rate(self) -> int:
+    def get_bonus_refund_rate(self) -> float:
```

Comment:
`float` is correct, but it may confuse, I mean there is only integer math in contract, but maybe it's ok

Reply:
Haha yeah, no the fence here myself.

Reply:
Maybe we should change the logic to be an integer division? Not necessarily in this PR, though.